drivers/gpio/gpio_pca95xx.c

```diff
@@ -0,0 +1,695 @@
+{
+	struct gpio_pca95xx_drv_data * const drv_data =
+		(struct gpio_pca95xx_drv_data * const)dev->driver_data;
+	union gpio_pca95xx_port_data *reg_out = &drv_data->reg_cache.output;
+	u32_t new_value;
+
+	new_value = reg_out->all & ~mask;
+
+	return gpio_pca95xx_write(dev, GPIO_ACCESS_BY_PORT, 0, new_value);
+}
+
+static int gpio_pca95xx_port_toggle_bits(struct device *dev, u32_t mask)
+{
+	struct gpio_pca95xx_drv_data * const drv_data =
+		(struct gpio_pca95xx_drv_data * const)dev->driver_data;
+	union gpio_pca95xx_port_data *reg_out = &drv_data->reg_cache.output;
+	u32_t new_value;
+
+	new_value = reg_out->all ^ mask;
+
+	return gpio_pca95xx_write(dev, GPIO_ACCESS_BY_PORT, 0, new_value);
+}
+
+static int gpio_pca95xx_pin_interrupt_configure(struct device *dev,
+						  unsigned int pin,
+						  enum gpio_int_mode mode,
+						  enum gpio_int_trig trig)
+{
+	return -ENOTSUP;
+}
+
+static const struct gpio_driver_api gpio_pca95xx_drv_api_funcs = {
+	.config = gpio_pca95xx_config,
+	.write = gpio_pca95xx_write,
+	.read = gpio_pca95xx_read,
+	.port_get_raw = gpio_pca95xx_port_get_raw,
+	.port_set_masked_raw = gpio_pca95xx_port_set_masked_raw,
+	.port_set_bits_raw = gpio_pca95xx_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_pca95xx_port_clear_bits_raw,
+	.port_toggle_bits = gpio_pca95xx_port_toggle_bits,
+	.pin_interrupt_configure = gpio_pca95xx_pin_interrupt_configure,
+};
+
+/**
+ * @brief Initialization function of PCA95XX
+ *
+ * @param dev Device struct
+ * @return 0 if successful, failed otherwise.
+ */
+static int gpio_pca95xx_init(struct device *dev)
+{
+	const struct gpio_pca95xx_config * const config =
+		dev->config->config_info;
+	struct gpio_pca95xx_drv_data * const drv_data =
+		(struct gpio_pca95xx_drv_data * const)dev->driver_data;
+	struct device *i2c_master;
+
+	/* Find out the device struct of the I2C master */
+	i2c_master = device_get_binding((char *)config->i2c_master_dev_name);
+	if (!i2c_master) {
+		return -EINVAL;
+	}
+	drv_data->i2c_master = i2c_master;
+
+	k_sem_init(&drv_data->lock, 1, 1);
+
+	return 0;
+}
+
+/* Initialization for PCA95XX_0 */
+#ifdef DT_INST_0_NXP_PCA95XX
+static const struct gpio_pca95xx_config gpio_pca95xx_0_cfg = {
+	.i2c_master_dev_name = DT_INST_0_NXP_PCA95XX_BUS_NAME,
+	.i2c_slave_addr = DT_INST_0_NXP_PCA95XX_BASE_ADDRESS,
+	.capabilities =
+		(DT_INST_0_NXP_PCA95XX_HAS_PUD ?  PCA_HAS_PUD : 0) |
+		0,
+};
+
+static struct gpio_pca95xx_drv_data gpio_pca95xx_0_drvdata = {
+	/* Default for registers according to datasheet */
+	.reg_cache.output = { .all = 0xFFFF },
+	.reg_cache.dir = { .all = 0xFFFF },
+	.reg_cache.pud_en = { .all = 0x0 },
+	.reg_cache.pud_sel = { .all = 0xFFFF },
+};
+
+/* This has to init after I2C master */
+DEVICE_AND_API_INIT(gpio_pca95xx_0, DT_INST_0_NXP_PCA95XX_LABEL,
+	    gpio_pca95xx_init,
+	    &gpio_pca95xx_0_drvdata, &gpio_pca95xx_0_cfg,
+	    POST_KERNEL, CONFIG_GPIO_PCA95XX_INIT_PRIORITY,
+	    &gpio_pca95xx_drv_api_funcs);
+
+#endif /* CONFIG_GPIO_PCA95XX_0 */
```

Comment:
```
#endif /* DT_INST_0_NXP_PCA95XX */
```

Reply:
Forgot about it... uploaded a new version.